### PR TITLE
Don't find services during tests.

### PR DESCRIPTION
**Pull request by hpidcock**
## Description of change

Prevent LXD container tests from querying services during tests.

## QA steps

Run tests.

## Documentation changes

N/A

## Bug reference

https://bugs.launchpad.net/juju/+bug/1881580

### CODE REVIEW

**Comment by hpidcock** [comment]
!!build!!

**Comment by manadart** [comment]
Updated description with LP bug. This needs to go to 2.7 as well.

File: container/lxd/initialisation_test.go [hunk 1]
```diff
@@ -336,7 +338,8 @@ func (s *InitialiserSuite) TestConfigureProxiesLXDNotRunning(c *gc.C) {
 	ctrl := gomock.NewController(c)
 	defer ctrl.Finish()
 	cSvr := lxdtesting.NewMockContainerServer(ctrl)
-	s.PatchForProxyUpdate(c, cSvr, false)
+	s.isLXDRunningLocally = false
```
Comment (manadart) [inline container/lxd/initialisation_test.go:341]: This won't work if `IsRunningLocally` is already patched in the setup, and `PatchForProxyUpdate` no longer patches that method.

Comment (hpidcock) [reply container/lxd/initialisation_test.go:341]: IsRunningLocally is patched with a function that returns the value s.isLXDRunningLocally. Setting it here changes it correctly.

Comment (manadart) [reply container/lxd/initialisation_test.go:341]: Right you are.

**Comment by wallyworld** [comment]
superseded by #11669

**Pull request closed without merging**